core/trino-main/src/main/java/io/trino/SystemSessionProperties.java

```diff
@@ -211,6 +212,11 @@ public SystemSessionProperties(
                         "Use preferred write partitioning",
                         featuresConfig.isUsePreferredWritePartitioning(),
                         false),
+                integerProperty(
+                        PREFERRED_WRITE_PARTITIONING_MIN_NUMBER_OF_PARTITIONS,
+                        "Use preferred write partitioning when number of written partitions exceeds given threshold`",
```

Comment:
Would it be possible to make the configuration toggle a little bit automatic?
I.e. how is the user going to determine whether they want to set this? 
If this depends e.g. on number of partitions vs number of nodes, we could encode this reasoning within server's logic

Reply:
The idea was to leave it backward-compatible (`0` as the default value) and let the user decide the threshold based on his/her workload if necessary.
However making this more "intelligent" is not a bad idea. The question is whether we want to do it here or as a follow-up.

Reply:
`when number of written` or `when the number of written`. I'm not native, but I think the latter is correct

Reply:
> If this depends e.g. on number of partitions vs number of nodes, we could encode this reasoning within server's logic

This doesn't really depend on number of workers. The hard limit is `hive.max-partitions-per-writers` which is independent of workers.